### ISSUE TITLE
No check for generic without constaints inside objects

### ISSUE THREAD

**Issue by val-o** [microsoft/TypeScript#43430]
# Bug Report

There is no check for type of argument in non-constrained generic if this argument is assigned to object with all fields optional

### 🔎 Search Terms

generics

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?jsx=0&ts=4.2.3&ssl=14&ssc=2&pln=1&pc=1#code/JYOwLgpgTgZghgYwgAgLIE8Aq6AOKDeAUMsgG5wA2ArhAFzL4BGA-PSFQLaPQC+hfhBAHsQAZzDI4yALzIAPJgB8AChxwocDvUwBKemiy4U0xQ2LIoEMFSggzJEuWp1kajR3N8BcZQCYdANyCIuLICAAWEAgA1vQY2Hgy9mSUNPS+-EA)

### 💻 Code

```ts
interface MyType {
  value: {b?: number}
}

const a = <T>(param: T):  MyType => {
  return {
    value: param
  }
}

a(2);
const check: MyType = {
  value: 2
}
```

### 🙁 Actual behavior

`a` allows me to put any type of variable as a param, even a primitive, whereas MyType[value] is only accepting objects

### 🙂 Expected behavior

TS should throw error saying that T is not assignable to {b?: number}

**Comment by RyanCavanaugh** [comment]
This is the intended behavior.
